meinberlin/apps/votes/forms.py

```diff
@@ -4,9 +4,29 @@
 from meinberlin.apps.votes.models import VotingToken
 
 
-class TokenForm(forms.Form):
+class VotingTokenWidget(forms.MultiWidget):
+
+    def __init__(self, attrs=None):
+        widgets = (
+            forms.TextInput(attrs={'size': '4', 'maxlength': '4'}),
+            forms.TextInput(attrs={'size': '4', 'maxlength': '4'}),
+            forms.TextInput(attrs={'size': '4', 'maxlength': '4'}),
+        )
+        super(VotingTokenWidget, self).__init__(widgets, attrs)
+
+    def decompress(self, value):
+        if value:
+            n = 4
+            return [value[i:i + n] for i in range(0, len(value), n)]
+        return [None, None, None]
 
-    token = forms.CharField(max_length=40)
+    def value_from_datadict(self, data, files, name):
+        values = super().value_from_datadict(data, files, name)
+        return "".join(values)
+
+
+class TokenForm(forms.Form):
+    token = forms.CharField(max_length=40, widget=VotingTokenWidget())
```

Comment:
The max_length validation doesn't work well with the three fields if it is shortened to 4. When it is shortened, each field really only allows 4 characters to be entered, but since we have a total of 12, this is in contradtion.